---
title: "ILRUN transcript expression"
author: "Marina Alexander"
date: "10/12/2019"
output: 
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}

# remind R where to look for libraries
.libPaths(c("C:/Users/ale097/Data School/Packages"))
# load libraries
library(tidyverse)
library(dplyr)
library(knitr)
library(rhdf5)

knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)
```

List the contents of the .h5 file
```{r list the contents}
h5ls("data/human_transcript_v7.h5")
```
Retrieve Gene Expression Omnibus info from matrix
```{r GEO accession}
samples = h5read("data/human_transcript_v7.h5", "meta/Sample_geo_accession")

samples %>% 
  as.tibble

```
Retreive sample information and tissue from matrix
```{r sample information}
tissue = h5read("data/human_transcript_v7.h5", "meta/Sample_source_name_ch1")

tissues <- tissue %>%
  as.tibble()
```



Join Geo acession and sample info
```{r joining GEO acession and sample info}
metadata <- samples %>%
  as.tibble() %>% 
  bind_cols(as.tibble(tissue)) %>% 
  rename(GEO_accession = value) %>% 
  rename(Tissue = value1) %>%
  filter(Tissue == "spleen"|
           Tissue == "lymph node"|
           Tissue == "adipose"|
           Tissue == "testes"|
           Tissue == "lung" |
           Tissue == "HeLa Cells"|
           Tissue == "Hela cells"|
           Tissue == "HeLa cells"|
           Tissue == "adipose_postLPS"|
           Tissue == "adipose tissue"|
           Tissue == "HeLa cell" |
           Tissue == "adipose, pre-LPS"|
           Tissue == "kidney"|
           Tissue == "cultured B-cells"|
           Tissue == "breast"|
           Tissue == "placenta"|
           Tissue == "Macrophages"|
           Tissue == "skeletal muscle"|
           Tissue == "CD4+ T cells"|
           Tissue == "T cells"|
           Tissue == "Liver"|
           Tissue == "NK cells"|
           Tissue == "Dentritic cells"|
           Tissue == "Neutrophils"|
           Tissue == "Plasma cell"|
           Tissue == "Thymus")



metadata
```



```{r extract ILRUN transcriptS}
transcripts = h5read("data/human_transcript_v7.h5", "meta/transcripts")
head(transcripts)

ilrun_transcripts <- transcripts %>% 
  as.tibble() %>% 
  rename(transcript_name = value) %>% 
  filter(transcript_name == "ENST00000374021.1"|
           transcript_name == "ENST00000374026.7"|
           transcript_name == "ENST00000374023.8")
```

```{r transcript expression}
samp = metadata$GEO_accession
sample_retrieve = which(samples %in% samp)
ilrun_expression = h5read("data/human_transcript_v7.h5", "data/expression", index = list(1:length(ilrun_transcripts$transcript_name), sample_retrieve))
H5close()
rownames(ilrun_expression) = ilrun_transcripts$transcript_name
colnames(ilrun_expression) = samples[sample_retrieve]

ilrun_expression %>% 
  as.tibble() %>% 
  head()

# Print file
extracted_ilrun_expression_file = "ilrun_expression_matrix.tsv"    
write.table(ilrun_expression, file=extracted_ilrun_expression_file, sep="\t", quote=FALSE)

print(paste0("Expression file was created at ", getwd(), "/", extracted_ilrun_expression_file))


```


```{r expression data}
# Identify columns to be extracted
samp = c("GSM1342291","GSM1505610","GSM1342284","GSM1342285","GSM1342286","GSM1342288","GSM1505585","GSM2343230","GSM2343387","GSM1010976","GSM1220585","GSM2343570","GSM1120317","GSM2344317","GSM1120316","GSM1335489","GSM2453445","GSM2453444","GSM2492500","GSM2492501","GSM2492502","GSM2492503","GSM2492508","GSM2492509","GSM2492510","GSM2492511","GSM2492516","GSM2492517","GSM2492518","GSM2492519","GSM2637980",
"GSM2637981","GSM2637982","GSM2637983","GSM2637984","GSM2637985","GSM2637986","GSM2637987","GSM2637988","GSM2637989","GSM2637990","GSM2637991","GSM2637992","GSM2637993","GSM2637994","GSM2637995","GSM2637996","GSM2637997","GSM2637998","GSM2637999","GSM2638000","GSM2638001","GSM2638002","GSM2638003","GSM2638004","GSM2638005","GSM2638006","GSM2638007","GSM2343541","GSM2343886","GSM2343888",
"GSM2344336","")

sample_locations = which(samples %in% samp)

# extract gene expression from compressed data
    expression = h5read("data/human_transcript_v7.h5", "data/expression", index=list(1:length(transcripts), sample_locations))
    H5close()
    rownames(expression) = transcripts
    colnames(expression) = samples[sample_locations]

    # Print file
extracted_expression_file = "Spleen_expression_matrix.tsv"    
write.table(expression, file=extracted_expression_file, sep="\t", quote=FALSE)
print(paste0("Expression file was created at ", getwd(), "/", extracted_expression_file))
    
```

```{r looking at ILRUN in spleen}

spleen <- read_tsv("Spleen_expression_matrix.tsv")


```


