---
title: "ILRUN transcript expression"
author: "Marina Alexander"
date: "10/12/2019"
output:
  pdf_document: default
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}

# remind R where to look for libraries
.libPaths(c("C:/Users/ale097/Data School/Packages"))
# load libraries
library(tidyverse)
library(dplyr)
library(knitr)
library(rhdf5)

knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)
```

List the contents of the .h5 file
```{r list the contents}
h5ls("data/human_transcript_v7.h5")

```
Retrieve Gene Expression Omnibus info from matrix
```{r GEO accession}
samples = h5read("data/human_transcript_v7.h5", "meta/Sample_geo_accession")

samples %>% 
  as.tibble

```
Retreive sample information and tissue from matrix
```{r sample information}
tissue = h5read("data/human_transcript_v7.h5", "meta/Sample_source_name_ch1")

```

Join Geo acession and sample info
```{r joining GEO acession and sample info}

metadata <- samples %>%
  as.tibble() %>% 
  bind_cols(as.tibble(tissue)) %>% 
  rename(GEO_accession = value) %>% 
  rename(Tissue = value1) %>%
  filter(Tissue == "Macrophages") %>% 
  filter(str_detect(GEO_accession, "GSM1896"))

 # to add in to code chunk above
          # Tissue == "lymph node"|
          # Tissue == "adipose"|
          # Tissue == "testes"|
          # Tissue == "lung" |
          # Tissue == "HeLa Cells"|
          # Tissue == "Hela cells"|
          # Tissue == "HeLa cells"|
          # Tissue == "adipose_postLPS"|
          # Tissue == "adipose tissue"|
          # Tissue == "HeLa cell" |
          # Tissue == "adipose, pre-LPS"|
          # Tissue == "kidney"|
          # Tissue == "cultured B-cells"|
          # Tissue == "breast"|
          # Tissue == "placenta"|
          # Tissue == "Macrophages"|
          # Tissue == "skeletal muscle"|
          # Tissue == "CD4+ T cells"|
          # Tissue == "T cells"|
          # Tissue == "Liver"|
          # Tissue == "NK cells"|
          # Tissue == "Dentritic cells"|
          # Tissue == "Neutrophils"|
          # Tissue == "Plasma cell"|
          # Tissue == "Thymus")


```



```{r extract ILRUN transcriptS}
transcripts = h5read("data/human_transcript_v7.h5", "meta/transcripts")
head(transcripts)

ilrun_transcripts <- transcripts %>% 
  as.tibble() %>% 
  rename(transcript_name = value) %>% 
  filter(transcript_name == "ENST00000374023.7"|
           transcript_name == "ENST00000374026.7"|
           transcript_name == "ENST00000374021.1")
```

```{r transcript expression in extracted tissue}
samp_ni = c("GSM1896082", "GSM1896085", "GSM1896088", "GSM1896091", "GSM1896094", "GSM1896097", "GSM1896100", "GSM1896103", "GSM1896106", "GSM1896112", "GSM1896115", "GSM1896118", "GSM1896121", "GSM1896124", "GSM1896127", "GSM1896130", "GSM1896109")

samp_inf = c("GSM1896083", "GSM1896086", "GSM1896089", "GSM1896092", "GSM1896095", "GSM1896098", "GSM1896101", "GSM1896104", "GSM1896107", "GSM1896110", "GSM1896113", "GSM1896116", "GSM1896119", "GSM1896122", "GSM1896125", "GSM1896128")

sample_retrieve_ni = which(samples %in% samp_ni)

sample_retrieve_inf = which(samples %in% samp_inf)

expression_ni = h5read("data/human_transcript_v7.h5", "data/expression", index = list(1:length(transcripts), sample_retrieve_ni))
H5close()
rownames(expression_ni) = transcripts
colnames(expression_ni) = samples[sample_retrieve_ni]

expression_inf = h5read("data/human_transcript_v7.h5", "data/expression", index = list(1:length(transcripts), sample_retrieve_inf))
H5close()
rownames(expression_inf) = transcripts
colnames(expression_inf) = samples[sample_retrieve_inf]

expression_ni %>% 
  as.data.frame()


  
```


```{r subsetting the data}
tissue_subset %>% 
  gather(sample, expression, -GSM1896130) %>% 
  arrange(desc(expression)) %>%
  rename(transcript_ID = GSM1896130)

ilrun <- tissue_subset %>% 
  gather(sample, expression, -GSM1896130) %>% 
  arrange(desc(expression)) %>%
  rename(transcript_ID = GSM1896130) %>%
  filter(transcript_ID == "ENST00000374023.7"| transcript_ID == "ENST00000374026.7", transcript_ID == "ENST00000374021.1")
  group_by(transcript_ID) %>% 
  summarise(mean = mean(expression), sd = sd(expression)) %>% 
  filter(mean >1) %>% 
  mutate(coeff_var = (sd/mean))


```


