---
title: "ILRUN transcript expression"
author: "Marina Alexander"
date: "10/12/2019"
output:
  pdf_document: default
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}

# remind R where to look for libraries
.libPaths(c("C:/Users/ale097/Data School/Packages"))
# load libraries
library(tidyverse)
library(dplyr)
library(knitr)
library(rhdf5)

knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)
```

List the contents of the .h5 file
```{r list the contents}
h5ls("data/human_transcript_v7.h5")


```
Retrieve Gene Expression Omnibus info from matrix
```{r GEO accession}
samples = h5read("data/human_transcript_v7.h5", "meta/Sample_geo_accession")

samples %>% 
  as.tibble

```
Retreive sample information and tissue from matrix
```{r sample information}
tissue = h5read("data/human_transcript_v7.h5", "meta/Sample_source_name_ch1")

tissues <- tissue %>%
  as.tibble()
```

Join Geo acession and sample info
```{r joining GEO acession and sample info}
metadata <- samples %>%
  as.tibble() %>% 
  bind_cols(as.tibble(tissue)) %>% 
  rename(GEO_accession = value) %>% 
  rename(Tissue = value1) %>%
  filter(Tissue == "Hela cells")

 # to add in to code chunk above
          # Tissue == "lymph node"|
          # Tissue == "adipose"|
          # Tissue == "testes"|
          # Tissue == "lung" |
          # Tissue == "HeLa Cells"|
          # Tissue == "Hela cells"|
          # Tissue == "HeLa cells"|
          # Tissue == "adipose_postLPS"|
          # Tissue == "adipose tissue"|
          # Tissue == "HeLa cell" |
          # Tissue == "adipose, pre-LPS"|
          # Tissue == "kidney"|
          # Tissue == "cultured B-cells"|
          # Tissue == "breast"|
          # Tissue == "placenta"|
          # Tissue == "Macrophages"|
          # Tissue == "skeletal muscle"|
          # Tissue == "CD4+ T cells"|
          # Tissue == "T cells"|
          # Tissue == "Liver"|
          # Tissue == "NK cells"|
          # Tissue == "Dentritic cells"|
          # Tissue == "Neutrophils"|
          # Tissue == "Plasma cell"|
          # Tissue == "Thymus")


```


No ilrun (ENST00000374023.8) or gapdh (ENST00000396861.5)???
```{r extract ILRUN transcriptS}
transcripts = h5read("data/human_transcript_v7.h5", "meta/transcripts")
head(transcripts)

ilrun_transcripts <- transcripts %>% 
  as.tibble() %>% 
  rename(transcript_name = value) %>% 
  filter(transcript_name == "ENST00000374021.1"|
           transcript_name == "ENST00000374026.7"|
           transcript_name == "ENST00000374023.8"|
           transcript_name == "ENST00000396861.5")
```

```{r transcript expression}

samp = metadata$GEO_accession
sample_retrieve = which(samples %in% samp)
hela_expression = h5read("data/human_transcript_v7.h5", "data/expression", index = list(1:length(transcripts), sample_retrieve))
H5close()
rownames(hela_expression) = transcripts
colnames(hela_expression) = samples[sample_retrieve]

# Print file
extracted_hela_expression_file = "hela_expression_matrix.tsv"    
write.table(hela_expression, file=extracted_hela_expression_file, sep="\t", quote=FALSE)

print(paste0("Expression file was created at ", getwd(), "/", extracted_hela_expression_file))

hela <- read_tsv("hela_expression_matrix.tsv")

hela %>% 
  gather(sample, expression, -GSM1048186) %>% 
  arrange(desc(expression)) %>%
  rename(transcript_ID = GSM1048186)

ilrun_hela <- hela %>% 
  gather(sample, expression, -GSM1048186) %>% 
  arrange(desc(expression)) %>%
  rename(transcript_ID = GSM1048186) %>% 
  filter(transcript_ID == "ENST00000374023.8")

ilrun_hela

```



